### DISCUSSION TITLE
Add custom SPI or extend keycloak SPI Credential or rest endpoint to get Credential

### DISCUSSION THREAD

**Discussion by SatriyoJati** [keycloak/keycloak#29003] [Q&A]
My goal is to make rest endpoint to get credential of user. My approach is using several class available in Keycloak such as JpaUserCredentialStore then it will return CredentialModel, then with this instance I can get value secret or using OTPCredentialModel to get TOTP secret data. I don't know whether it is the  correct way or not. Do you have any suggestion whether I should use those class or create custom SPI or not?

**Comment by codespearhead** [comment]
What use case justifies that feature? Because it seems like you're trying to create a malicious Keycloak extension.
